### PR TITLE
converts redis logger to exception handler

### PR DESCRIPTION
**Pull request by db3dev**
### PR Checklist
- [x] All `TODO`'s are done and removed
- [x] `package.json` has correct information
- [x] All dependencies are of correct type (regular vs peer vs dev)
- [x] Documented any complicated or confusing code
- [x] No linter errors
- [x] Project packages successfully (`npm pack`)
- [x] Installed local packed version in another project and tested
- [x] Updated README

#### Changes proposed in this pull request:

**JIRA Task Link:**

### CODE REVIEW

File: src/modules/services/redis/redis.service.ts [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
-        private readonly logger: log4js.Logger,
+        @Inject()
+        private readonly logger: ErrorHandler
     ) {
         // tslint:disable
-        this.client.on('error', (err) => this.logger.error('Error:', err));
-        this.client.on('ready', () => this.logger.info('Connected to Redis'));
-        this.client.on('reconnecting', () => this.logger.info('Attempting to reconnect to Redis...'));
+        this.client.on('error', (err) => this.logger.captureException(err));
+        this.client.on('ready', () => this.logger.captureMessage('Connected to Redis'));
```
Comment (jpinkster) [inline src/modules/services/redis/redis.service.ts]: For connecting, can this just be a breadcrumb? Don’t want to flog Sentry with it connecting successfully

File: src/modules/services/redis/redis.service.ts [hunk 2]
```diff
@@ -1,22 +1,23 @@
...
+        this.client.on('reconnecting', () => this.logger.captureMessage('Attempting to reconnect to Redis...'));
+        this.client.on('end', () => this.logger.captureException(new RedisException(new Error('Redis Connection Fatal'))))
```
Comment (jpinkster) [inline src/modules/services/redis/redis.service.ts]: What do we do here?

**Comment by tvmaasjazz** [comment]
I just updated the redis provider to have the configs we discussed as well as handling the killing of the client and then creating a new one. I left a couple of TODOs in there that are questions to see if you have an opinion on those. Also, I feel pretty good about how I'm doing things structurally but this a new pattern so let me know if you have any suggestions or something looks weird. @jpinkster

File: src/modules/providers/redis/redis.provider.ts [hunk 3]
```diff
@@ -1,28 +1,103 @@
...
+     *    can't connect for 10 seconds
+     * 3. Subscribe and emit the new connection's events
+     * 4. Return the client
+     */
+    setupClient() {
+        // initial connection
+        const self = this;
+        const client = redis.createClient({
+            retry_strategy(options: any) {
+                // TODO: maybe have this be 5 seconds before a new client created? (feels like if it's
```
Comment (jpinkster) [inline src/modules/providers/redis/redis.provider.ts]: yeah just an arbitrary number ... make a judgement call and run with it

File: src/modules/providers/redis/redis.provider.ts [hunk 4]
```diff
@@ -1,28 +1,103 @@
...
+                    self.emit('error', new Error('Retry time exhausted, creating new client connection...'));
+
+                    // cleanly end connection, then setup new client connection
+                    client.quit();
+                    self.connection = self.setupClient();
+                    return;
+                }
 
-const client = bluebird.promisifyAll(redis.createClient(redisConfig)) as redis.RedisClient;
+                // TODO: this was in the example... but maybe just keep trying every 100ms for the 10 seconds?
```
Comment (jpinkster) [inline src/modules/providers/redis/redis.provider.ts]: no opinion ... your call here .. sorry lol

**Pull request merged**